### ISSUE TITLE
"Type dosnt have "OptionsConstructor" [E016]"

### ISSUE THREAD

**Issue by sandorvasas** [typegoose/typegoose#478]
Terminal Error:
```
Type dosnt have "OptionsConstructor" [E016]#
Error: Type does not have an valid "OptionsConstructor"! (${getName(loggerType)} on ${getName(target)}.${pkey}) [E016]
```

Documentation Error Details:
```
Details:
The Type provided does not have an property OptionsConstructor (undefined / null), this is required to map options of an array & map
```

What the f does that even mean? The "Details" section basically contains the same sentence as the error message, only the word order is different, but there's no additional meaning to it. If the user doesn't understand the error message  (and I don't), then they won't understand the details either. What should be done if this error happens, how to fix this?

**Comment by hasezoey** [comment]
please provide the correct error message, the first one seems to be copied directly from code

and please provide what on what type/property this error occurred on (code example)

**Comment by sandorvasas** [comment]
I'm trying to get embedded schema arrays to work. e.g.
```ts
@modelOptions({})
export class Photo  {
  readonly _id: ObjectId;
  @prop()
  thumbnail: string
  @prop()
  fileName: string
}
export const PhotoModel = getModelForClass(Photo)

@prop({ default: [], get: v=>v||[], set: v=>v, type: Photo })
photos: Photo[]
```
I also tried `type: PhotoModel`, doesn't work. With previous typegoose (@hasezoey/typegoose), it worked well just by having the `Photo` there, without defining a `type` in `@prop`.

**Comment by hasezoey** [comment]
> it worked well just by having the Photo there, without defining a type in @prop.

you need to define the option `type` if using an Map or Array, because of #300 

---

does 31042dff1e4cc4110497717a1b97d9dc293bb7b0 clarify some things?

**Comment by hasezoey** [comment]
should be fixed with 9d0806c5c943a2928c42126a3f6dfd41c0d1f3c5, which is included in 7.4.8

**Comment by sandorvasas** [comment]
Using mongoose 5.10 and @types/mongoose 5.10, declaring
```
mongoose.Schema.Types.Custom.OptionsConstructor = mongoose.SchemaTypeOptions; // prefer this if possible
```

says there is no "Custom" defined on mongoose.Schema.Types.Also there is no `SchemaTypeOptions` defined on mongoose. What mongoose version should I be using?

**Comment by hasezoey** [comment]
`SchemaTypeOptions` and `OptionsConstructor` (still) dont exist in the mongoose types (both 5.10 and 5.11)

> says there is no "Custom" defined on mongoose.Schema.Types

that workaround is, as the documentation says `If custom types are used`

and an `Nested Class` (Your Case) is not an custom type

and as i wrote earlier, this issue should be fixed with 7.4.8

**Comment by sandorvasas** [comment]
It _is_ fixed, just tried it, compiles fine. Thanks!
